.azure-pipelines/windows.sh

```diff
@@ -0,0 +1,160 @@
+    install_host_dmd
+else
+    echo 'Invalid $D_COMPILER provided'.
+    exit 1
+fi
+
+################################################################################
+# Checkout other repositories
+################################################################################
+
+REPO_BRANCH="$SYSTEM_PULLREQUEST_TARGETBRANCH"
+
+for proj in druntime phobos; do
+    if [ "$REPO_BRANCH" != master ] && [ "$REPO_BRANCH" != stable ] &&
+            ! git ls-remote --exit-code --heads "https://github.com/dlang/$proj.git" "$REPO_BRANCH" > /dev/null; then
+        # use master as fallback for other repos to test feature branches
+        clone "https://github.com/dlang/$proj.git" "${DMD_DIR}/../$proj" master
+        echo "[GIT_CLONE] Switched $proj to branch master \$(REPO_BRANCH=$REPO_BRANCH)"
+    else
+        clone "https://github.com/dlang/$proj.git" "${DMD_DIR}/../$proj" "$REPO_BRANCH"
+        echo "[GIT_CLONE] Switched $proj to branch $REPO_BRANCH"
+    fi
+done
+
+################################################################################
+# Prepare build flags
+################################################################################
+
+export CC="${MSVC_CC}"
+export AR="${MSVC_AR}"
+
+if [ "$MODEL" == "64" ] ; then
+    export MODEL_FLAG="-m64"
+    MAKE_FILE="win64.mak"
+else
+    export MODEL_FLAG="-m32"
+    MAKE_FILE="win32.mak"
+fi
+
+################################################################################
+# Build DMD
+################################################################################
+
+DMD_BIN_PATH="$DMD_DIR/generated/windows/release/${MODEL}/dmd"
+
+cd "${DMD_DIR}/src"
+"${DM_MAKE}" -f "${MAKE_FILE}" reldmd DMD="$DMD_BIN_PATH"
+
+################################################################################
+# WORKAROUND: Make the paths to CC and AR whitespace free
+# REASON: Druntime & Phobos Makefiles as the variables don't use quotation
+################################################################################
+
+ln -s "$(dirname "$CC")" "${DMD_DIR}/../ccdir"
+export CC="$DMD_DIR/../ccdir/cl.exe"
+export AR="$DMD_DIR/../ccdir/lib.exe"
+
+################################################################################
+# WORKAROUND: Build zlib separately with DigitalMars make
+# REASON: whitespace path variables in DigitalMars make from indirect invocation from Phobos
+################################################################################
+
+cd "${DMD_DIR}/../phobos/etc/c/zlib"
+${DM_MAKE} -f win64.mak "MODEL=${MODEL}" "zlib${MODEL}.lib" CC="${CC}" LIB="${AR}" VCDIR="${VCINSTALLDIR}"
+
+################################################################################
+# Build Druntime and Phobos
+################################################################################
+
+for proj in druntime phobos; do
+    cd "${DMD_DIR}/../${proj}"
+    "${DM_MAKE}" -f "${MAKE_FILE}" DMD="$DMD_BIN_PATH" CC="$CC" AR="$MSVC_AR" LD="$MSVC_LD" VCDIR="${VCINSTALLDIR}" CFLAGS="/C7"
```

Comment:
The build stumbles because MSVC_LD no longer exists.

Reply:
Hmm, I didn't mean that MSVC_LD should be reintroduced, but that it does not have to be passed on the command line. $LD is not anywhere (but in some unused targets of the zlib makefiles).